lib/usd/translators/meshWriter.cpp

```diff
@@ -41,21 +42,178 @@
 #include <pxr/usd/usdSkel/root.h>
 #include <pxr/usd/usdUtils/pipeline.h>
 
+#include <maya/MFloatArray.h>
+#include <maya/MFnBlendShapeDeformer.h>
 #include <maya/MFnDependencyNode.h>
+#include <maya/MFnGeometryFilter.h>
 #include <maya/MFnMesh.h>
+#include <maya/MFnSkinCluster.h>
+#include <maya/MGlobal.h>
 #include <maya/MIntArray.h>
+#include <maya/MItDependencyGraph.h>
 #include <maya/MPlug.h>
 #include <maya/MPlugArray.h>
 #include <maya/MString.h>
 #include <maya/MStringArray.h>
 #include <maya/MUintArray.h>
 
+#include <cfloat>
 #include <set>
 #include <string>
 #include <vector>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+MStatus mayaDisableAllBlendShapesForMesh(
+    const MObject& mesh,
+    MObjectArray&  blendShapeNodes,
+    MFloatArray&   envelopes)
+{
+    MStatus status;
+    // NOTE: (yliangsiew) If there's a skinCluster, find that first since that
+    // will be the intermediate to the blendShape node. If not, just search for any
+    // blendshape deformers upstream of the mesh.
```

Comment:
Hi, thanks for the review!

I think the code currently checks if there is a skinCluster; if not, it will
start the search from the mesh instead, so it should be finding all blendshape
deformers upstream regardless. This is the same code being used in
`mayaFindOrigMeshFromBlendShapeTarget`.